addons/gdUnit4/src/core/parse/GdScriptParser.gd

```diff
@@ -307,7 +307,7 @@ func tokenize_value(input: String, current: int, token: Token) -> Token:
 	var next := 0
 	var current_token := ""
 	# test for '--', '+-', '*-', '/-', '%-', or at least '-x'
-	var test_for_sign := (token == null or token.is_operator()) and input[current] == "-"
+	var test_for_sign := (token == null or token.is_operator()) and input[current] == "-" as bool
```

Comment:
This solves this error:
```
SCRIPT ERROR: Parse Error: Could not infer the type of the variable "test_for_sign" because the initial value is a variant. Use explicit "Variant" type if this is intended.
```

Reply:
This is also strange, which is not necessary to set as bool.
This line is a full boolean operation and will default to type bool.